core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -439,6 +443,8 @@ final class ShuffleBlockFetcherIterator(
                   fetchRequests += FetchRequest(address, Array((blockId, size)))
                   result = null
                 }
+                bytesInFlight -= size
+                fetchUpToMaxBytes()
```

Comment:
we may throw exception before hitting this branch

Reply:
Thanks @cloud-fan 
IIUC, i think when  `fetchfailedException` thrown there is no need to consider this.Which is also the original design.